### ISSUE TITLE
Four boring, though useful "administrative" suggestions

### ISSUE THREAD

**Issue by llaalleexxaanndd** [Karry/osmscout-sailfish#319]
1) Navigation: Make it possible to select route From by selection in the Collections
2) Scale/measurement blob/circle in the centre of the screen
3) Local geoid wgs84 height "error" compensation
4) Advanced search

TLDR:
1) Sometimes it may be convenient to plan a route without physically (gps wise) being at the start location. It can be cumbersome to manually select "from" and "to" locations by moving the map around and selecting or even necessary by copying the gps location from the map to paste it into the search dialogue in the navigation menu, in case you need to specify both a distant "to" and "from" location on the map. I think it would be helpful (at least to me) if it is possible to select a location already saved in the collections through the navigation menu. I.e. an addition to the "Current location, Search, Pick a place"-drop down menu with "Select from Collections".
I know that you can select "Route to" by holding a waypoint until the drop down menu appears. An addition of a "route from" from this menu would help, but I think it will be more logical to add it the other way around through the navigation menu.
I also know that you can "hack" this by modifying the waypoint distance by adding some additional zeros, but if could be navigated though the collections menu, a lot of scrolling and grief could be avoided (I think).

2) The ruler in the lower left hand corner is already great, but sometimes you may want (at least I do) to get a quick rough view of distance in other directions than East-West. And you have to move endpoints down into the corner to make the measurement. I think that a (on/off switchable, like the collections star) big semi-transparent blob/circle in the centre of the map with the same diameter/scale as the length of  the ruler in the corner would be a big help to give a quick view of the direct distance between two points on the map, in any direction.

3) I find that (at least in my location) the height indicator is misleading. I am not to say if this deviation is a constant, lineary or worse, but as far as I have read, this is due to the "inaccuracy" of the wgs84 geoid model which in turn is based on the earth's gravitation pull at a given location, which in turn is of course depending of the distance to the centre of the Earth, but also of the density of the material between you and the centre. Worse in some parts of the world than others. 40 m is no a big deal if you are in a country with real mountains, but in a country where all "mountains" are below 200 m and you are never far away from the sea (and level) the deviation is very much in your face :-). The best way would of course be if the compensation was implemented directly in the OS, perhaps as an extension to the local Mozilla Location Services, so all applications would benefit from it. However as Jolla most likely has bigger fish to fry, this will likely happen never than sooner. A dirty (but perhaps not quite so quick as I think) solution could be to be able to add a compensation constant (hopefully a constant will be all it takes rather than an a and a b for a linear ax+b function or even worse). It could be added by letting the user input the value (positive or negative) for the constant in the downloaded maps info menu, perhaps? So that there would be a chance that it would give a better approximation of the actual height for each area you are located in, when using offline maps, based on the height reported by the gps with the compensation constant (or function :-O).

4) Advanced search: Free text search for stuff is fine, but I find that it often gives too wide results in a seemingly random order (if you wait long enough). Can be frustrating, when you actually know, sort of, what you are looking for. I think it would be beneficial if it was possible to give the search engine specific hints to what you are looking for. I would love (at least I think so) if there was an advanced search menu option, where it was possible to specify (in text or by drop down) e.g. country (if only from list based on present downloaded maps), city name, street name, number, amenity, shop etc., "within distance".

Not exactly rocket science additions, but hopefully not too hard to implement either (I think).

Thank you for your time.

**Comment by Karry** [comment]
Hi @llaalleexxaanndd , thank you for your suggestions! I would say that these are not boring. I will create separate issues from individual points 1,2 and 4. I have some of them on my private TODO list already, but it will be better to track them publicly.

To point 3 - what do you mean with height indicator? You mean info panel with height above sea from gps, right? It may be inaccurate, but adding manual compensation is too geeky IMHO. I cannot imagine how I would calibrate my height...

**Comment by llaalleexxaanndd** [comment]
Hi Karry! 

And thank you for looking into the suggestions, likewise. 

Height above sea level, yes, exactly. But too nerdy?... Perhaps, but with an appropriate user interface for it, I am sure it could become totally mainstream. :-) 

I have noticed that tapping the GPS height above (mean) sea level indicator in the upper left corner already opens the "search for nearby natural peak pois" 
poi:10000:natural_peak 

This already makes it quite easy to find a nearby survey point with a by definition correct height above sea level indication. If there are any. Otherwise the height could be known from other sources e.g. printed maps or a GPS with "magic" geoid correction. I do not know how "they" do this. 

A suggestion (which may be improved): Touching and holding the GPS height indicator in the upper left corner opens a menu: Geoid correction, where user may enter the correct height, which the app then uses to calculate the correction value and store it as a correction to the area covered by the downloaded map. Message: Correction will be applied to <"whatever offline map that covers your current location">. The menu should also have the option to clear the value. 

I hope you will take it into consideration. It is rather annoying to me, when i.e. GPS Info tells me that the vertical accuracy is as low as 3 m and the reported altitude at the same time if off by 40 m. 
Not that a correction in OSMScout makes any change to what the GPS Info app reports, but at least OSMScout's output will be more on point, especially an enhancement when you are using the tracker and share the track with 3rd parties. 

About the blob/circle for quick and dirty distance measurement "as the crow flies": Great, whatever is the most interesting and challening, but not too demanding function to implement. 
Thought the concern about mercator projection as a source of error also already applies to the ruler in the lower left hand corner and apparently you are changing the scale of the ruler depending on latitude (and longitude)*)? Anyway the easy fix for the blob would simply to disable it for the lower zoom levels...? 

Thank you for your time. 

Yours 
llaalleexxaanndd 
*) at least, "funny" things happen when you are zoomed out to "world view" and e.g. place Australia right under the ruler and move the map around and pinch in and out. 

PS: (I am the crazy person who has downloaded 13+ offline maps, issue #273. I eventually hacked a shell script to manage/enable/disable/hide the offline maps. For entertainment, I have added the script and output to the issue). 




Fra: "Lukáš Karas" ***@***.***> 
Til: "Karry/osmscout-sailfish" ***@***.***> 
Cc: "llaalleexxaanndd" ***@***.***>, "Mention" ***@***.***> 
Sendt: torsdag, 5. oktober 2023 00:21:57 
Emne: Re: [Karry/osmscout-sailfish] Four boring, though useful "administrative" suggestions (Issue #319) 





Hi [ https://github.com/llaalleexxaanndd | @llaalleexxaanndd ] , thank you for your suggestions! I would say that these are not boring. I will create separate issues from individual points 1,2 and 4. I have some of them on my private TODO list already, but it will be better to track them publicly. 

To point 3 - what do you mean with height indicator? You mean info panel with height above sea from gps, right? It may be inaccurate, but adding manual compensation is too geeky IMHO. I cannot imagine how I would calibrate my height...

**Comment by Karry** [comment]
With the height error correction, it reminds me, that some Garmin (?) devices are using barometer for precise elevation changes. Not for absolute value, but just for changes in small time interval (hours). It would be nice to be able use barometer sensor on Sailfish OS... 

Anyway, are you sure that gps height error is always the same? I am worry that thing you are writing about is just generic gps error that may be "jumping" in few minutes... When the error is changed significantly during one trip, it would be pain to use such correction...

**Comment by llaalleexxaanndd** [comment]
Yes, I think it is Garmin and others that use a barometer to measure changes. I think I even tested a cycle computer once a long time ago which used this method. Not a great use case in this country, this island and region where the highest point is 40 km away and only around 120 m above sea level and you otherwise will be pressed to find anything above 80 m. 

I am quite sure that it is not spurious GPS results causing the "errors". My current location is always reported in the neighborhood of 59-63 m above sea level, while a survey map shows the actual height is more like 20 m. I have several tracks created sailing (sail boat / motor boat track classification type for tracks wanted :-), and though there are outliers, they seem to gravitate around 35-40 m above sea level. A track of a 3.5 hour bicycle tour around the centre of Copenhagen a month ago gravitates around "slightly below" 50 m above sea level. No statistical analysis made, though. 

I will go and "survey" and visit the nearby highpoint at 67 m above sea level a few times and send you the tracks, if you are interested or if it may make you interested in taking this interesting matter further. 

Thank you for your time 

There is an if not interesting, then at least very infomative 10 year old thead (use archive.org to read the many useful links in the thread) on reddit called "How is a 0' sea level established for GPS measurements? My floating boat is currently reading 141' over sea level!" where many seemingly knowledgeable redditors give their take on survey datum, geoid models &c.
